board/aspeed/evb_ast2700/evb_ast2700.c

```diff
@@ -213,7 +213,7 @@ int set_board_info(const u8* scm_eeprom_buf, const u8* hpm_eeprom_buf)
 	else {
 		bin2hex(mac_str, scm_eeprom_buf + LAST_2B_MAC0_ADDR, LAST_2B_MAC0_LEN);
 		old_bootargs = env_get(ENV_BOOTARGS);
-		if  ( !(strstr(old_bootargs, "systemd.hostname")) ) {
+		if  ( (get_platform_name(board_id, &plat_name) == 0) && (!(strstr(old_bootargs, "systemd.hostname"))) ) {
```

Comment:
Avoid complex conditions and get board_id in its own "if" block.

Reply:
Done. Changed platform name fetching as topmost if statement.